Default to false for warehouse perm on API

gigxz opened the pull request:
During local testing I observed that `user.can_edit_users?` can return `nil`, which causes an API failure because the field is non-nullable. Update it to default to false.

I am not clear on what situation causes this method to return null. I _think_ it's when the user has **no roles**. I have only seen it in rspec testing, I haven't replicated it in development.


Branch name is for testing https://github.com/greenriver/hmis-frontend/pull/1121

gigxz committed:
change for build

gigxz committed:
Default value for perm

gigxz committed:
fail fast

gigxz committed:
rm fail fast

gigxz commented:
@eanders maybe this should be fixed higher up the chain somewhere. It does seem odd that `can_<perm>?` would ever return nil, don't you think? Or is that intentional?

eanders commented:
> @eanders maybe this should be fixed higher up the chain somewhere. It does seem odd that `can_<perm>?` would ever return nil, don't you think? Or is that intentional?

I don't think that's intentional.  I'm surprised it hasn't come up until now.
I believe that _should_ be solvable by changing:


```
    define_method(permission) do
      @permissions ||= load_effective_permissions.merge(load_health_effective_permissions)
      @permissions[permission]
    end
```

to 

```
    define_method(permission) do
      @permissions ||= load_effective_permissions.merge(load_health_effective_permissions)
      @permissions[permission] || false
    end
```

Does that seem like it might work?

gigxz committed:
Merge branch 'release-158' into gig/system-ci-fix

gigxz committed:
Accept suggestion, add regression test case

gigxz commented:
@eanders that worked great! I updated it and also added a regression test case https://github.com/greenriver/hmis-warehouse/pull/5272/commits/abd8f420dd2fc7a4fa164b190d5d47fac7de1a59

eanders reviewed (approved):
Thanks @gigxz !
